### PR TITLE
Add a possible Appveyor configuration with matrix, caching, etc.

### PR DESCRIPTION
**Pull request by padraic**
Implements #24, though you'll have to sign up to the service yourself!

If it's not immediately known, Appveyor is basically Travis CI but running on Windows. Can help detect any weirdness on that platform which developers still insist on using (so I hear :grin:). Like Travis, it's free for open source projects so long as you stay within its cache limits, which should not be an issue unless you have too many builds added to the matrix.

I've set it to use typical composer requirements - you can add a `--prefer-lowest` set of of builds if desired at at a later point by editing the matrix.

### CODE REVIEW

**Comment by raphaelstolt** [comment]
Should be export-ignored in .gitattributes.
```text
appveyor.yml export-ignore
```

File: .gitattributes [hunk 1]
```diff
@@ -7,3 +7,4 @@ LICENSE export-ignore
 phpunit.xml.dist export-ignore
 README.md export-ignore
 tests/ export-ignore
+appveyor.yml export-ignore
```
Comment (povils) [inline .gitattributes]: newline

**Comment by povils** [comment]
To be honest appveyor.yml file looks a mess ;/

**Comment by padraic** [comment]
It is messy, unfortunately there's no PHP automatically installed so you have to DIY the installation. On the plus side, the caching section usually shaves about 3 minutes off the build times.

**Comment by povils** [comment]
Thanks. Please add new line in .gitattributes ad i'll merge this one :)

**Pull request merged**